### ISSUE TITLE
Net Core build for BSD

### ISSUE THREAD

**Issue by ta264** [Radarr/Radarr#5828]
It would be nice to add a net core build to our build pipeline.

@Thefrank if you wanted to help out that would be great.  Also see our discord https://discord.gg/r5wJPt9

**Comment by Thefrank** [comment]
I advise against this simply because there is no official Microsoft support for dotnet5 under any BSD system. There is community support but dotnet5 will likely not see an "Official" Microsoft supported version. NET6...maybe...if we can get everything from 5 working 100% with 0 test failures. 

Mono is still supported under FreeBSD. Someone was nice enough to get Mono 6.8.0 setup as both pkg and ports.

That being said, if you still want to go ahead with it, let me know and I can provide whatever support I can give. You mentioned a missing NuGet package; I will try and get some of the more commonly needed ones added to the public link ASAP. I am also taking suggestions on non-mega.nz places that might be better to upload all of these sdks and binaries.

I ran `dotnet test ./src/Radarr.sln --runtime freebsd-x64 -f net5.0 --results-directory /root/_testresults --logger:html /nr:false` and got the following results. Most of the failures seem to be not-running-under-win32 or missing mediainfo. For the later, I don't have it installed in my jail that I used for CI/CD

[Radarr-aa00959010e01ab0df1e32c195bcdffa8488080e-Results.tar.gz](https://github.com/Radarr/Radarr/files/5876458/Radarr-aa00959010e01ab0df1e32c195bcdffa8488080e-Results.tar.gz)

I'll take a look at a build too hopefully a bit later

**Comment by Thefrank** [comment]
welp, good news bad news.
`dotnet publish ./src/Radarr.sln -f net5.0 -c Release --self-contained true -r freebsd-x64 /nr:False` produces a working binary for FreeBSD!
```
root@building:~/testrad/freebsd-x64 # ldd ./Radarr
./Radarr:
        libthr.so.3 => /lib/libthr.so.3 (0x800683000)
        libc++.so.1 => /usr/lib/libc++.so.1 (0x8006b0000)
        libcxxrt.so.1 => /lib/libcxxrt.so.1 (0x80077d000)
        libm.so.5 => /lib/libm.so.5 (0x80079f000)
        libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x8007d1000)
        libc.so.7 => /lib/libc.so.7 (0x80024e000)
```
bad news: it looks like it still wants mono bits:
```
root@building:~/testrad/freebsd-x64 # ./Radarr --help
[Info] Bootstrap: Starting Radarr - /root/testrad/freebsd-x64/Radarr.dll - Version 10.0.0.27749 


[Fatal] ConsoleApp: EPIC FAIL! 

[v10.0.0.27751] TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Common.Instrumentation.InitializeLogger
 ---> TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Common.EnvironmentInfo.IOsInfo
 ---> TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Common.EnvironmentInfo.OsInfo
 ---> TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Common.EnvironmentInfo.IOsVersionAdapter
 ---> TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Mono.EnvironmentInfo.VersionAdapters.IssueFileVersionAdapter
 ---> TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Common.Disk.IDiskProvider
 ---> TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Mono.Disk.DiskProvider
 ---> TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Mono.Disk.ICreateRefLink
 ---> TinyIoC.TinyIoCResolutionException: Unable to resolve type: NzbDrone.Mono.Disk.RefLinkCreator
 ---> System.IO.FileNotFoundException: Could not load file or assembly 'Mono.Posix.NETStandard, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.

File name: 'Mono.Posix.NETStandard, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
   at NzbDrone.Mono.Disk.RefLinkCreator..ctor(Logger logger)
   at lambda_method4(Closure , Object[] )
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3576
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3583
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3503
   at TinyIoC.TinyIoCContainer.SingletonFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2742
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3237
   at TinyIoC.TinyIoCContainer.Resolve(Type resolveType, String name) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 1458
   at NzbDrone.Common.Composition.Container.<>c__DisplayClass12_0.<CreateSingletonImplementationFactory>b__0(TinyIoCContainer c, NamedParameterOverloads p) in /root/Radarr/src/NzbDrone.Common/Composition/Container.cs:line 85
   at TinyIoC.TinyIoCContainer.DelegateFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2504
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.DelegateFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2508
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3237
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3551
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3564
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3503
   at TinyIoC.TinyIoCContainer.SingletonFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2742
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3237
   at TinyIoC.TinyIoCContainer.Resolve(Type resolveType, String name) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 1458
   at NzbDrone.Common.Composition.Container.<>c__DisplayClass12_0.<CreateSingletonImplementationFactory>b__0(TinyIoCContainer c, NamedParameterOverloads p) in /root/Radarr/src/NzbDrone.Common/Composition/Container.cs:line 85
   at TinyIoC.TinyIoCContainer.DelegateFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2504
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.DelegateFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2508
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3237
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3551
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3564
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3503
   at TinyIoC.TinyIoCContainer.SingletonFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2742
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3237
   at TinyIoC.TinyIoCContainer.Resolve(Type resolveType, String name) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 1458
   at NzbDrone.Common.Composition.Container.<>c__DisplayClass12_0.<CreateSingletonImplementationFactory>b__0(TinyIoCContainer c, NamedParameterOverloads p) in /root/Radarr/src/NzbDrone.Common/Composition/Container.cs:line 85
   at TinyIoC.TinyIoCContainer.DelegateFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2504
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.DelegateFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2508
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3237
   at TinyIoC.TinyIoCContainer.<>c__DisplayClass134_0.<ResolveAllInternal>b__2(TypeRegistration registration) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3665
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.CastIterator[TResult](IEnumerable source)+MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at NzbDrone.Common.EnvironmentInfo.OsInfo..ctor(IEnumerable`1 versionAdapters, Logger logger) in /root/Radarr/src/NzbDrone.Common/EnvironmentInfo/OsInfo.cs:line 51
   at lambda_method1(Closure , Object[] )
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3576
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3583
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3503
   at TinyIoC.TinyIoCContainer.SingletonFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2742
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3237
   at TinyIoC.TinyIoCContainer.Resolve(Type resolveType, String name) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 1458
   at NzbDrone.Common.Composition.Container.<>c__DisplayClass12_0.<CreateSingletonImplementationFactory>b__0(TinyIoCContainer c, NamedParameterOverloads p) in /root/Radarr/src/NzbDrone.Common/Composition/Container.cs:line 85
   at TinyIoC.TinyIoCContainer.DelegateFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2504
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.DelegateFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 2508
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3237
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3551
   --- End of inner exception stack trace ---
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3564
   at TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3508
   at TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 3328
   at TinyIoC.TinyIoCContainer.Resolve(Type resolveType) in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 1431
   at TinyIoC.TinyIoCContainer.Resolve[ResolveType]() in /root/Radarr/src/NzbDrone.Common/TinyIoC.cs:line 1550
   at NzbDrone.Common.Composition.Container.Resolve[T]() in /root/Radarr/src/NzbDrone.Common/Composition/Container.cs:line 36
   at Radarr.Host.Bootstrap.Start(StartupContext startupContext, IUserAlert userAlert, Action`1 startCallback) in /root/Radarr/src/NzbDrone.Host/Bootstrap.cs:line 35
   at NzbDrone.Console.ConsoleApp.Main(String[] args) in /root/Radarr/src/NzbDrone.Console/ConsoleApp.cs:line 41


Press enter to exit...
```

**Comment by ta264** [comment]
Thanks a lot. The libmonoposixhelper I can fix, you can see build instructions here if you want to try: https://github.com/Servarr/libMonoPosixHelper/blob/master/build_libmonohelper.sh

The resulting .so needs putting in `src/Runtimes/freebsd-x64` in the radarr source tree.

I'll have a go with your updated link thanks. I had to edit `/usr/share/dotnet/sdk/5.0.102/Microsoft.NETCoreSdk.BundledVersions.props` to include freebsd-x64 rid to convince it to build at all using your SDK bundle. Not sure if something needs tweaking in the bundle...

**Comment by Thefrank** [comment]
> include freebsd-x64 rid
I knew I forgot to do something. It's also just a simple `sed` statement to add it but I still forgot! I will upload a fixed version with those added.

**Comment by ta264** [comment]
I have a working build based on your 5.0.102 sdk.  The 5.0.103 version wants a newer AspNetCore.App.Runtime nuget.  I think maybe you're building based on the `release/5.0` branch rather than the `v5.0.2` tag (the latest current release) so there's a version mismatch - there's no release of 5.0.103 yet.

The `bsd-net-core` branch contains the compiled `libMonoPosixHelper.so`.  The error you had is caused by a wonky `Radarr.Mono.deps.json`.  If you replace the `Mono.Posix.NETStandard` section with
```
      "Mono.Posix.NETStandard/5.20.1-preview": {
        "runtime": {
          "runtimes/linux-x64/lib/netstandard2.0/Mono.Posix.NETStandard.dll": {
            "assemblyVersion": "1.0.0.0",
            "fileVersion": "1.0.0.0"
          }
        },
        "native": {
          "runtimes/linux-x64/native/libMonoPosixHelper.so": {
            "fileVersion": "0.0.0.0"
          }
        }
      },
```
then it starts and runs, at least for me.  I need to look into what's wrong with this package - it's always been a thorn in my side!

**Comment by ta264** [comment]
@Thefrank I have a working build, but it seems your SDK is built on FreeBSD 12 so the resulting binaries don't work on FreeBSD 11.

Any chance you could provide the SDK built on FreeBSD 11?  Or give me a quick overview of how to get it compiled myself?

**Comment by Thefrank** [comment]
TL;DR: might be mlock issue that is not fixable otherwise if its ABI or kernel version related it can be fixed but will just annoy me to do so hah 


Does it give a CoreCLR error about memory exhaustion?

It _should_ work on vanilla FreeBSD releng/11.4 outside of a jail or VM but I only ever tested dotNET 3.0 and 3.1 on it
It will likely not work any FreeNAS (aka anything FreeBSD 11.2-RELEASE and earlier variant) due to a lack of adding this upstream patch: https://docs.freebsd.org/cgi/getmsg.cgi?fetch=145846+0+archive/2019/svn-src-head/20190519.svn-src-head
It absolutely works on TrueNAS (FreeBSD 12.2-RELEASE) both in and out of VMs.

Does it give an ABI error? Does it give a kernel version error?

This is fixable but annoying and I opted for just building it under 12 because stable/11 is already getting close to EOL and most of the releng/11.x are already past it

**Comment by ta264** [comment]
Running the published app fails with `Undefined symbol "statfs@FBSD_1.5"`.  Poking around I think libc had `statfs@FBSD_1.0` or 1.1 in 11.4.  This is not my area of expertise though so could be on the wrong track.

I was hoping using an SDK build on 11 this would enable it to work on FreeNAS releases since I think that's our main BSD userbase.  But if that's not possible then maybe it's not worth wasting time, and we drop support for anything before TrueNAS.

**Comment by Thefrank** [comment]
Very odd.  
Is that the full error? Also can you give me the output of `ldd ./Radarr` ?
It hopefully looks something like this:
```
# ldd ./Radarr
./Radarr:
        libthr.so.3 => /lib/libthr.so.3 (0x800684000)
        libc++.so.1 => /usr/lib/libc++.so.1 (0x8006b1000)
        libcxxrt.so.1 => /lib/libcxxrt.so.1 (0x80077e000)
        libm.so.5 => /lib/libm.so.5 (0x8007a0000)
        libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x8007d2000)
        libc.so.7 => /lib/libc.so.7 (0x80024e000)
```

**Comment by ta264** [comment]
The full output is:
```
root@freebsd-s-4vcpu-8gb-nyc3-01:~/Radarr # ./Radarr -nobrowser
/root/Radarr/libcoreclr.so: Undefined symbol "statfs@FBSD_1.5"
```
and ldd:
```
root@freebsd-s-4vcpu-8gb-nyc3-01:~/Radarr # ldd ./Radarr
./Radarr:
        libthr.so.3 => /lib/libthr.so.3 (0x801000000)
        libc++.so.1 => /usr/lib/libc++.so.1 (0x801228000)
        libcxxrt.so.1 => /lib/libcxxrt.so.1 (0x8014f2000)
        libm.so.5 => /lib/libm.so.5 (0x801711000)
        libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x801941000)
        libc.so.7 => /lib/libc.so.7 (0x800825000)
```
This was built on a FreeBSD 12.2 VM then the output bundle copied to a 11.4 VM for testing.  It worked fine copied to a second 12.2 VM.

**Comment by Thefrank** [comment]
ok, looks like I should build the SDK for 11 (which _should_ also work on 12)
one last thing before I start: can you post the output of `ldd ./libcoreclr.so`? just want to make sure im not missing something

Something I missed from earlier post: I do build from `release/5.0` and I dont remember exactly why. I want to say that as one point the tag was not correctly syncing across runtime/aspnetcore/sdk/installer but that doesnt seem right.

RantNotReallyRelatedToThis: FreeNAS 11 (FreeBSD 11.x) is basically EOL as iXSystems is moving everyone over to TrueNAS. FreeBSD is EOL on everything under releng/11.4. FreeBSD stable/12 is under active support and stable/13 should be coming in a few months. The reason I did not build for stable/11 was that the common FreeBSD variant (FreeNAS) was already past EOL and its upstream parts had less than a year of support before EOL. If you _really really_ want to support FreeBSD releng/11.4 you can and I have no issues rebuilding the SDK (when I get some free time to babysit a build...this weekend likely) but you might be inviting a lot of questions from FreeNAS users about updating jails, setting jail params, manually installing tarballs, manually installing dependences, and more. You might still get these by targeting stable/12 but at least with that there is a very small chance that a community plugin for Radarr will be made for TrueNAS so you won't get the above questions.

**Comment by ta264** [comment]
Sorry, I've killed the VM, will have to dig that out tomorrow.  Happy to help out with the build if you provide some instructions :) Managed to cross build the runtime from linux for bsd11 but couldn't figure out the SDK.

Re the rant, some input from someone that's knowledgeable on BSD/FreeNAS would be super helpful but perhaps you can jump on the discord and we can talk it over.  Personally I find this sort of discussion much easier in real-time.

**Comment by Thefrank** [comment]
I will try and get on Discord later
I was able to get the published Radarr working from `bsd-net-core` branch. I ended up having to copy the `Mono.Posix.NETStandard.dll` from the cached NuGet package into the finished product (along with the helper...and the deps changes) but it runs now

**Comment by Thefrank** [comment]
Works in build environment now

![Screenshot 2021-01-27 152320](https://user-images.githubusercontent.com/1910378/106062215-549a3b00-60ab-11eb-9389-8d3c4efbdaf8.png)

I was going to attach the bundle but its 85MB and I can only do 10MB here so its on mega

**Comment by ta264** [comment]
Yeah that package is missing a freebsd-x64 runtime and looks like freebsd doesn't fall back to Linux so it doesn't manage to resolve it at all.  I ended up repacking it with an added runtime then the publish command goes off without a hitch.

I'm GMT so it'll be tomorrow for me now. Hopefully catch you at some point :)

**Comment by Thefrank** [comment]
@ta264 close because finished? or wait until next release/milestone?

- [ ] Crossbuild SDK for FreeBSD 11 as FreeBSD 12 exclusive SDK already exists
- [ ] Update script for FreeBSD support
- [ ] Build Radarr under Linux targeting FreeBSD
- [ ] Build Radarr natively under FreeBSD
- [ ] 100% pass on tests in FreeBSD
- [ ] (Optional and unlikely to produce a result) bug port maintainer to update port

**Comment by mike-pt** [comment]
I know this is closed, but curious if there are any steps on:

`Build Radarr natively under FreeBSD` 

I clone the develop branch locally and followd https://github.com/jasonpugsley/installer/wiki/.NET-5.0-Preview-3-for-FreeBSD to thet donet working, so I got the tarballs for dotnet preview and the packages, updated nuget config (both local and on radarr src/NuGet.confg so it point to the local pkgs 

However when I do 

```
./build.sh --backend --enable-bsd
```

I get 
```
Start 'Build'
Failed to create CoreCLR, HRESULT: 0x8007FF02
```

**Comment by bakerboy448** [comment]
There's no need to build it yourself. BSD binaries are provided within the build stream now.

For BSD support/questions/further discussion swing by our 
[Discord](https://radarr.video/discord) in the #bsd-builds channel
